public/app/plugins/datasource/cloud-monitoring/__mocks__/cloudMonitoringQuery.ts

```diff
@@ -28,12 +35,17 @@ export const createMockSLOQuery: (overrides?: Partial<SLOQuery>) => SLOQuery = (
   };
 };
 
-export const createMockQuery: (overrides?: Partial<CloudMonitoringQuery>) => CloudMonitoringQuery = (overrides) => {
+export const createMockQuery: (overrides?: Subset<CloudMonitoringQuery>) => CloudMonitoringQuery = (overrides) => {
```

Comment:
curious: what's the difference between Partial and Subset? Is it to maintain the original type?

Reply:
`Partial` doesn't work for nested objects (so in this case if we pass this override to `createMockQuery`: `createMockQuery({ metricQuery: { projectName: '$testVar' } })` then we will get a type error as `Partial` allows `metricQuery` to be undefined but this does not extend to the properties within `metricQuery`. `Subset` extends `Partial` to work for nested objects in this case 😊